doc/examples/filters/plot_deconvolution.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+"""
+=====================
+Image Deconvolution
+=====================
+In this example, we deconvolve an image using Richardson-Lucy
+deconvolution algorithm.
+Richardson-Lucy deconvolution
+-----------------------------
+The algorithm is based on Bayesian-based iterative method
+where an image blurred by a point-spread function is 
+recovered. 
+.. [1] William Hadley Richardson, "Bayesian-Based Iterative 
+       Method of Image Restoration",
+       J. Opt. Soc. Am. A 27, 1593-1607 (1972)
```

Comment:
Please, add a reference to wikipedia for the reader who doesn't know the details: https://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution